src/item.h

```diff
@@ -704,8 +704,9 @@ class item : public JsonSerializer, public JsonDeserializer
         /**
          * Callback when a player starts wielding the item. The item is already in the weapon
          * slot and is called from there.
+         * @param mv number of moves *already* spent wielding the weapon
          */
-        void on_wield( player &p );
+        void on_wield( player &p, int mv = 0 );
```

Comment:
> Having those available in on_wield looks like it could result in spaghetti code with no clear rules what handles what.

That's partly why I wanted to avoid `on_wield` returning anything but I take your point that the callback itself updates the player moves

> Maybe the wield function should be split
> would just checking move difference be too hacky here?

Possibly, but it isn't clear at present how to best split the various parts

> The current solution is probably good enough (given the complicate situation). And it's documented, so there shouldn't be a problem.

If you don't object can we revisit this next time any changes are made to `player::wield`, `player::wield_contents` and `item::on_wield` as there might be more context to push us in a particular direction?